src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -2023,7 +2047,7 @@ private ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedA
             var parameterListList = ArrayBuilder<ImmutableArray<ParameterSymbol>>.GetInstance();
             foreach (var m in methods)
             {
-                if (!IsUnboundGeneric(m) && m.ParameterCount > 0)
+                if (!IsUnboundGeneric(m) && m.GetParametersIncludingExtensionParameter().Length > 0)
```

Comment:
>GetParametersIncludingExtensionParameter [](http://example.com/codeflow?start=46&length=40)

It might be better to have an API that returns combined count and avoids unnecessary allocations.